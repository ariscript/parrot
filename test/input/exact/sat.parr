def even(n):
    if n == 0: true
    else: if n == 1: false
    else: odd(n - 1)
and def odd(n):
    if n == 0: false
    else: if n == 1: true
    else: even(n - 1)

check:
    10 satisfies even,
    42 violates odd,
    11 satisfies odd,
    83 violates even.

0
