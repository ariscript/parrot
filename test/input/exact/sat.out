check block: 4/4 tests passed
0
